docs/docs/mlflow-3/index.mdx

```diff
@@ -0,0 +1,348 @@
+            """User-defined functions (UDFs) in the context of Databricks and Apache Spark are custom functions that you can create to perform specific tasks on your data. These functions are written in a programming language such as Python, Java, Scala, or SQL, and can be used to extend the built-in functionality of Spark.\n\nUDFs can be used to perform complex data transformations, data cleaning, or to apply custom business logic to your data. Once defined, UDFs can be invoked in SQL queries or in DataFrame transformations, allowing you to reuse your custom logic across multiple queries and applications.\n\nTo use UDFs in Databricks, you first need to define them in a supported programming language, and then register them with the SparkSession. Once registered, UDFs can be used in SQL queries or DataFrame transformations like any other built-in function.\n\nHere\'s an example of how to define and register a UDF in Python:\n\n```python\nfrom pyspark.sql.functions import udf\nfrom pyspark.sql.types import IntegerType\n\n# Define the UDF function\ndef multiply_by_two(value):\n    return value * 2\n\n# Register the UDF with the SparkSession\nmultiply_udf = udf(multiply_by_two, IntegerType())\n\n# Use the UDF in a DataFrame transformation\ndata = spark.range(10)\nresult = data.withColumn("multiplied", multiply_udf(data.id))\nresult.show()\n```\n\nIn this example, we define a UDF called `multiply_by_two` that multiplies a given value by two. We then register this UDF with the SparkSession using the `udf` function, and use it in a DataFrame transformation to multiply the `id` column of a DataFrame by two.""",
+        ],
+    }
+)
+
+# Start a run to represent the evaluation job
+with mlflow.start_run() as evaluation_run:
+    eval_dataset = mlflow.data.from_pandas(
+        df=eval_df, name="eval_dataset", targets="expected_response"
+    )
+    # Run the evaluation based on extra metrics
+    result = mlflow.evaluate(
+        model=f"models:/{logged_model.model_id}",
+        data=eval_dataset,
+        # targets="expected_response",
+        model_type="question-answering",
+        extra_metrics=[
+            mlflow.metrics.latency(),
+            mlflow.metrics.genai.answer_correctness("openai:/gpt-4o-mini"),
+        ],
+        # This is needed since answer_correctness looks for 'inputs' field
+        evaluator_config={"col_mapping": {"inputs": "messages"}},
+    )
+    # Log evaluation metrics and associate with agent
+    mlflow.log_metrics(
+        metrics=result.metrics,
+        dataset=eval_dataset,
+        # Specify the ID of the agent logged above
+        model_id=logged_model.model_id,
+    )
```

Comment:
Do we need this? Does `mlflow.evaluate` automatically link the metrics to the model?